### PR TITLE
Use native GH actions cancel feature

### PR DESCRIPTION
**Pull request by fjetter**
This uses the GH actions native cancel feature which is currently in beta, see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency

The benefit is that this does not require is to spawn another job to trigger the cancel and is therefore more reliable in case the CI is flooded

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@ name: Tests
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{github.ref}}
```
Comment (crusaderky) [inline .github/workflows/tests.yaml:6]:
```suggestion
  group: ${{ github.ref }}
```

**Review by crusaderky — APPROVED** [review]
LGTM. Approved conditionally to producing evidence that it works as intended.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -2,6 +2,10 @@ name: Tests
 
 on: [push, pull_request]
 
+concurrency:
```
Comment (crusaderky) [inline .github/workflows/tests.yaml:5]:
```suggestion
# When this workflow is queued, automatically cancel any previous running
# or pending jobs from the same branch
concurrency:
```

**Comment by fjetter** [comment]
Tested this on my fork. Here you can see an example where the jobs are cancelled. https://github.com/fjetter/distributed/actions/runs/1011571451

The message then shows a message like `Canceling since a higher priority waiting request for 'refs/heads/cancel-runs' exists` where `refs/heads/cancel-runs` is the branch name / ref as exposed to `${{ github.ref }}`

I'll commit your suggestions and we should see this work on this PR, already.

**Comment by fjetter** [comment]
Tests were successfully canceled after the last commit https://github.com/dask/distributed/actions/runs/1011583254 was replaced with https://github.com/dask/distributed/actions/runs/1011722431

**Pull request merged**